lib/open_project/journal_formatter/project_phase_definition.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+# -- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+# ++
+
+class OpenProject::JournalFormatter::ProjectPhaseDefinition < JournalFormatter::NamedAssociation
+  private
+
+  def associated_object_name(object)
+    if phase_active?(object)
+      super
+    else
+      "#{super} (#{I18n.t(:label_inactive)})"
+    end
+  end
+
+  def phase_active?(definition)
+    phase = @journal.journable.project.phases.detect { |phase| phase.definition_id == definition.id } if definition
```

Comment:
Why load all phases and not find needed phase by `definition_id`?

Reply:
It is a bet. While fetching that one specific phase via a find_by directly would definitely be faster for a single journal, I wager that the performance will increase if the phase is changed more than once. Since the project is shared across journals, so are the phases. By that, journals can access them again without taking a performance hit.

Reply:
Ok, let it be